app/controllers/hq/users_controller.rb

```diff
@@ -6,4 +6,24 @@ def index
     @users = User.all
   end
 
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+    if @user.update_attributes(user_params)
+      flash[:success] = 'User successfully saved'
+      redirect_to hq_user_url(@user)
+    else
+      render 'edit'
+    end
+  end
+
+private
+
+  def user_params
+    params[:user].permit(:user_name, :email)
```

Comment:
Including `require` shortcuts the process so that it fails before going to the model. Additionally, the failure is clearer: the model will report that a required attribute is missing, but it won't be immediately obvious why. With `required`, the controller will throw `ActionController::ParameterMissing`, making it clear that the form failed to submit a required parameter.

If you look at #252, you'll see that there was some discussion and disagreement about this and a related issue. The Partners controller does not use `require` because of a certain aspect of its architecture that the Users controller does not replicate.